src/com/facebook/buck/ide/intellij/IjModuleGraphFactory.java

```diff
@@ -315,6 +325,36 @@ public static IjModuleGraph from(
     return new IjModuleGraph(depsBuilder.build());
   }
 
+  private static IjModule createExtraModuleForCompilerOutput(
+      IjModule module, IjProjectConfig projectConfig, ProjectFilesystem projectFilesystem) {
+    Path projectRootPath = Paths.get(projectConfig.getProjectRoot());
+    Path extraModuleBasePath = projectRootPath
+        .resolve("buck-out")
+        .resolve("extra-intellij-output")
+        .resolve(projectRootPath.relativize(module.getModuleBasePath()));
+
+    Path extraModuleRelativePath = projectRootPath.relativize(extraModuleBasePath.normalize());
+    if (!projectFilesystem.exists(extraModuleRelativePath)) {
+      try {
+        projectFilesystem.mkdirs(extraModuleRelativePath);
```

Comment:
we are creating a synthetic library module so that we can use it's compiler output path to put the r class files on the class path for the Layout Preview. the module is created based off of the base path, which, might not already exist for a clean build. if it doesn't, then the `IjProjectTemplateDataPreparer` runs into a `NoSuchFileException` when it expects the synthetic library module's base path to exist. this prevents that from occurring